app/models/lesson.rb

```diff
@@ -57,6 +58,11 @@ def delete_associated_asl_files
     FileUtils.remove_dir "#{Rails.root}/public/storylines/#{id}", true
   end
 
+  def delete_associated_user_completions
+    completions = CompletedLesson.where(lesson_id: self.id)
```

Comment:
I'm not sure you need the `self` on this, as I think it's implicit. Rubocop might catch it as a style thing, but no other negative things I can think of than "style"